src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private Random random = new Random();
+    private ColorSupplier supplier = new ColorSupplier();
+    private Figure isoscelesTrapezoid = new IsoscelesTrapezoid(supplier.getRandomColor(),
+            random.nextDouble(), random.nextDouble(), random.nextDouble());
+    private Figure rectangle = new Rectangle(supplier.getRandomColor(),
+            random.nextDouble(), random.nextDouble(), random.nextDouble());
+    private Figure circle = new Circle(supplier.getRandomColor(),
+            random.nextDouble());
+    private Figure rightTriangle = new RightTriangle(supplier.getRandomColor(),
+            random.nextDouble(), random.nextDouble());
+    private Figure square = new Square(supplier.getRandomColor(), random.nextDouble());
+    private Figure [] figures = {isoscelesTrapezoid, rectangle, circle, rightTriangle, square};
```

Comment:
You should avoid creating instances of different figures as class fields in FigureSupplier. Instead, create these instances inside the method where they are needed. This way, you can pass random values to the constructor of a figure instead of generating them inside figure classes, improving code readability and avoiding tight coupling between the FigureSupplier and the specific figures.